File: API/Entities/Member.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Entities;

public class Member
{
    public string Id { get; set; } = string.Empty;
    public DateOnly BirthDate { get; set; }
    public string? ImageUrl { get; set; }
    public required string DisplayName { get; set; }
    public DateTime CreatedOn { get; set; } = DateTime.UtcNow;
    public DateTime LastActive { get; set; } = DateTime.UtcNow;
    public required string Gender { get; set; }
    public string? Description { get; set; }
    public required string City { get; set; }
    public required string Country { get; set; }


    //navigation properties
    public ICollection<Photo> Photos { get; set; } = [];

    [ForeignKey(nameof(Id))]
    public AppUser AppUser { get; set; } = null!;
}

File: client/src/core/models/Member.ts
export interface Member {
  id: string;
  birthDate: string;
  imageUrl?: string | null;
  displayName: string;
  createdOn: string;
  lastActive: string;
  gender: string;
  description?: string | null;
  city: string;
  country: string;
  photos: Photo[];
}

export interface UpdateMember {
  displayName: string;
  description?: string | null;
  city: string;
  country: string;
}

export interface Photo {
  id: number;
  url?: string | null;
  isMain: boolean;
}
